src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt

```diff
@@ -342,4 +344,35 @@ class PrisonServiceTest {
         .isInstanceOf(ContactDetailsNotFoundException::class.java)
     }
   }
+
+  @Nested
+  inner class UpdateContactDetails {
+
+    private val prisonRepository2: PrisonRepository = mock()
+    private val contactDetailsRepository2: ContactDetailsRepository = mock()
+
+    @Test
+    fun `should throw ContactDetailsNotFoundException when getByPrisonIdAndType contactDetail not found`() {
+      val removeIfNull = true
+      val updateRequest = ContactDetailsDto(VIDEOLINK_CONFERENCING_CENTRE, emailAddress = "xxx@moj.gov.uk", phoneNumber = "01234567899", webAddress = "www.xxxmojdigital.blog.gov.uk")
+      val prison = Prison("BRI", "Bri Prison", active = true)
+
+      whenever(prisonRepository.getReferenceById(any())).thenReturn(prison)
```

Comment:
you are using _prisonRepository_ but declared _ prisonRepository2_ is that on purpose?

Reply:
why do you need to create new variables?

Reply:
prisonRepository2 is used in the next Test, line 367.  When you use mocks they can retain the previos state and i got a different behaviour result so I considered create a new one.

Reply:
I have added a commit to reset repository mocks beforeEach this fix the issue of reusing them. this apply to all 3 PR opened with similar tests.